test/AssetHolder/transferAll.test.ts

```diff
@@ -8,146 +8,106 @@ import {
-    ${description2}  | ${2}  | ${true}    | ${'2'} | ${[A]}     | ${['1']}      | ${[]}     | ${[]}        | ${[]}       | ${[]}            | ${['1']}      | ${'1'}    | ${undefined}
-    ${description3}  | ${3}  | ${true}    | ${'1'} | ${[A]}     | ${['2']}      | ${[A]}    | ${['1']}     | ${[]}       | ${[]}            | ${['1']}      | ${'0'}    | ${undefined}
-    ${description4}  | ${4}  | ${true}    | ${'1'} | ${[C]}     | ${['1']}      | ${[]}     | ${[]}        | ${[C]}      | ${['1']}         | ${[]}         | ${'0'}    | ${undefined}
-    ${description5}  | ${5}  | ${true}    | ${'2'} | ${[C]}     | ${['1']}      | ${[]}     | ${[]}        | ${[C]}      | ${['1']}         | ${[]}         | ${'1'}    | ${undefined}
-    ${description6}  | ${6}  | ${true}    | ${'1'} | ${[C]}     | ${['2']}      | ${[C]}    | ${['1']}     | ${[C]}      | ${['1']}         | ${[]}         | ${'0'}    | ${undefined}
-    ${description7}  | ${7}  | ${true}    | ${'2'} | ${[A, B]}  | ${['1', '1']} | ${[]}     | ${[]}        | ${[]}       | ${[]}            | ${['1', '1']} | ${'0'}    | ${undefined}
-    ${description8}  | ${8}  | ${true}    | ${'1'} | ${[A, B]}  | ${['1', '1']} | ${[B]}    | ${['1']}     | ${[]}       | ${[]}            | ${['1']}      | ${'0'}    | ${undefined}
-    ${description9}  | ${9}  | ${true}    | ${'3'} | ${[A, B]}  | ${['2', '2']} | ${[B]}    | ${['1']}     | ${[]}       | ${[]}            | ${['2', '1']} | ${'0'}    | ${undefined}
-    ${description10} | ${10} | ${true}    | ${'2'} | ${[C, X]}  | ${['1', '1']} | ${[]}     | ${[]}        | ${[C, X]}   | ${['1', '1']}    | ${[]}         | ${'0'}    | ${undefined}
-    ${description11} | ${11} | ${true}    | ${'1'} | ${[C, X]}  | ${['1', '1']} | ${[X]}    | ${['1']}     | ${[C]}      | ${['1']}         | ${[]}         | ${'0'}    | ${undefined}
-    ${description12} | ${12} | ${true}    | ${'3'} | ${[C, X]}  | ${['2', '2']} | ${[X]}    | ${['1']}     | ${[C, X]}   | ${['2', '1']}    | ${[]}         | ${'0'}    | ${undefined}
+    name                              | heldBefore            | setOutcome      | newOutcome | heldAfter             | payouts         | reason
+    ${' 0. outcome not set         '} | ${{c: 1}}             | ${{}}           | ${{}}      | ${{}}                 | ${{A: 1}}       | ${reason}
+    ${' 1. funded          -> 1 EOA'} | ${{c: 1}}             | ${{A: 1}}       | ${{}}      | ${{}}                 | ${{A: 1}}       | ${undefined}
+    ${' 2. overfunded      -> 1 EOA'} | ${{c: 2}}             | ${{A: 1}}       | ${{}}      | ${{c: 1}}             | ${{A: 1}}       | ${undefined}
+    ${' 3. underfunded     -> 1 EOA'} | ${{c: 1}}             | ${{A: 2}}       | ${{A: 1}}  | ${{}}                 | ${{A: 1}}       | ${undefined}
+    ${' 4. funded      -> 1 channel'} | ${{c: 1, C: 0}}       | ${{C: 1}}       | ${{}}      | ${{c: 0, C: 1}}       | ${{}}           | ${undefined}
```

Comment:
Do we need the `C: 0` here? I think it's important that it works without that too.

Reply:
Addressed in bfde6ba
![Screenshot 2019-09-11 at 15 35 18](https://user-images.githubusercontent.com/1833419/64706687-cf0fee00-d4a9-11e9-8a7d-c5e67e30561c.png)